### PR TITLE
Switch from teams to collaborators

### PR DESCRIPTION
**Pull request by pholleran**
### TL;DR

This PR modifies the action to audit a repository's collaborators instead of its teams.

### Why?

1. Many organizations require a static list of permissions for auditing. The present use of `teams` doesn't meet this need as:

* a separate lookup is required to determine which people are on which team
* team membership is dynamic, which would require organizations to also capture the membership of each team at the time of each action run

2. Outside collaborators and individual organization members added to the repository are not captured when auditing repository teams. 

3. the `listCollaborators` endpoint returns all collaborators, whether added as a member of an organization team, member of the org, or an outside collaborator

### What this PR does

* Modifies the action logic to get collaborators instead of teams
* Groups log output by repository using `core.startGroup()` and `core.endGroup()`
* Updates the README

The resulting JSON object is much longer, but provides a complete audit of all users with access to repos as well as their access levels.

### CODE REVIEW

**Comment by gr2m** [comment]
I do not remember all the details, it's been a while since I created the action. But I do know that the repository collaborators endpoint did not include the `maintain` and `triage` roles until very recently, that's why we didn't use them when we created the action initially.

We don't have any tests right now so I'm happy to trust you on this one. Do you use the action yourself? Did you verify that it works for you? /cc @jeffwilcox

**Comment by pholleran** [comment]
@gr2m - I'm looking to use it and have tested it in an org of ~220 people. Happy to take a stab at writing some tests.

If this gets accepted I'm also thinking of PRing an optional SAMLidentities flag that will map `user.login` with a SAML identity if the org has SAML enabled as well as a different PR to provide a .csv output option.

Essentially, I'm looking to bring in a few features from [https://github.com/svanboxel/org-audit-action](svanboxel/org-audit-action) and combine them with the more efficient queries and app-based auth structure of this Action. I figured it to be a better approach than building yet another audit action 😄 

I've verified that both the `repos/{owner}/{repo}/collaborators` and `repos/{owner}/{repo}/teams` return the same permissions object, below, despite the current state of the API docs:

```json
    "permissions": {
      "admin": true,
      "maintain": true,
      "push": true,
      "triage": true,
      "pull": true
    }
```

**Comment by gr2m** [comment]
Did you run your fork of the action? I just want to make sure you confirmed that it works the way it's documented and that you expect. If so, we can merge in your changes.

In terms of the test setup, you can compare to the tests at https://github.com/gr2m/twitter-together/, that works pretty well for me. But feel free to try something else if you have other preferences.
